src/native/libs/System.Native/pal_networkchange.c

```diff
@@ -35,6 +35,14 @@ Error SystemNative_CreateNetworkChangeListenerSocket(int32_t* retSocket)
     sa.nl_family = AF_NETLINK;
     sa.nl_groups = RTMGRP_LINK | RTMGRP_IPV4_IFADDR | RTMGRP_IPV4_ROUTE | RTMGRP_IPV6_ROUTE;
     int32_t sock = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+	
+	// Added receive timeout to prevent recvmsg method in SystemNative_ReadEvents to block thread indefinitely.
+	// This allows method SystemNative_ReadEvents to periodically exit and allows .NET thread to exit if needed.
+	struct timeval tv;
+	tv.tv_sec = 1; //timeout in seconds
+	tv.tv_usec = 0;
+	setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, (const char*)&tv, sizeof tv);
+	
```

Comment:
tabs --> spaces
```suggestion

    // Added receive timeout to prevent recvmsg method in SystemNative_ReadEvents to block thread indefinitely.
    // This allows method SystemNative_ReadEvents to periodically exit and allows .NET thread to exit if needed.
    struct timeval tv;
    tv.tv_sec = 1; //timeout in seconds
    tv.tv_usec = 0;
    setsockopt(sock, SOL_SOCKET, SO_RCVTIMEO, (const char*)&tv, sizeof tv);

```

Reply:
Also this should probably go after the return file descriptor check on lines 49-53, i.e. somewhere after:
```C
    if (sock == -1)
    {
        *retSocket = -1;
        return (Error)(SystemNative_ConvertErrorPlatformToPal(errno));
    }
```

And if the `setsockopt` is necessary for this to work, there should probably be return value check similar to what we have with `bind`.

Reply:
You are right, I will add this after socket was successfully created and then also check setsockopt result.